config/ldes-delta-pusher/handle-regular-types.ts

```diff
@@ -0,0 +1,80 @@
+import { Changeset } from "../types";
+import { query } from "mu";
+import { publish } from "./publisher";
+import { log } from "./logger";
+
+type InterestingSubject = {
+  uri: string;
+  ldesType: string;
+};
+
+const regularTypesToLDESMapping = {
+  "http://data.vlaanderen.be/ns/mandaat#Fractie": "fractie",
+  "http://www.w3.org/ns/org#Membership": "lidmaatschap",
+  "http://data.vlaanderen.be/ns/mandaat#Mandaat": "mandaat",
+  "http://www.w3.org/ns/person#Person": "persoon",
+  "http://purl.org/dc/terms/PeriodOfTime": "periode",
+  "http://purl.org/dc/terms/Identifier": "identifier",
+  "http://data.vlaanderen.be/ns/persoon#Geboorte": "geboorte",
+};
+
+const keepRegularTypesQuery = async (subjects: string[]) => {
+  const types = Object.keys(regularTypesToLDESMapping);
+  const matches = await query(`
+      SELECT DISTINCT ?s ?type WHERE {
+        ?s a ?type .
+        VALUES ?type { ${types.map((type) => `<${type}>`).join(" ")} }
+        VALUES ?s { ${[...subjects]
+          .map((subject) => `<${subject}>`)
+          .join(" ")} }
+      }
+    `);
+  return matches.results.bindings
+    .map((binding) => {
+      const ldesType = regularTypesToLDESMapping[binding.type.value];
+      if (!ldesType) {
+        return null;
+      }
+      return { uri: binding.s.value, ldesType };
+    })
+    .filter((b) => !!b) as InterestingSubject[];
+};
+
+const keepRegularTypeSubjects = async (changesets: Changeset[]) => {
+  const subjects = new Set<string>();
+  for (const changeset of changesets) {
+    changeset.inserts.forEach((insert) => {
+      subjects.add(insert.subject.value);
+    });
+    changeset.deletes.forEach((del) => {
+      subjects.add(del.subject.value);
+    });
+  }
+
+  const allSubjects = [...subjects];
+  let subjectsToKeep: InterestingSubject[] = [];
+  const chunkSize = 1000;
```

Comment:
Isn't this pretty big as chunk size, since this means the values of the subject could be 1000 different values in the query? But I'm no export on mu-cl-resources, this could be perfectly reasonable and pretty efficient.